word_language_model/main.py

```diff
@@ -130,14 +157,24 @@ def evaluate(data_source):
     model.eval()
     total_loss = 0.
     ntokens = len(corpus.dictionary)
-    hidden = model.init_hidden(eval_batch_size)
+    if args.model == 'Transformer':
+        tgt_mask = model.transformer.generate_square_subsequent_mask(args.bptt).to(device)
```

Comment:
Using member variables (`model.transformer`) directly is, in my opinion, not a good idea, because you might want to change them. You could either create a setter, or another method that relays it or hide this altogether as suggested in other comments.